### DISCUSSION TITLE
How to put 3js object into hubs' room?

### DISCUSSION THREAD

**Discussion by DurianBoomb** [mozilla/hubs#2826] [General Questions & Discussion] [answered]
Hello everyone! My project has encountered some problems in key places. I wrote a music visualization file with 3js, but I don't know how to put it in the room and make it work. Can you give me some suggestions?
![image](https://user-images.githubusercontent.com/34919698/89707954-be425380-d96a-11ea-9d92-aeddf8527e6f.png)

**Comment by emclaren** [comment, marked as answer]
Hi there, unfortunately there is nowhere to add custom code into a room on hubs.mozilla.com yet (hopefully this is something we will be able to add in the future)

It is possible on Hubs Cloud -  see this response: 
https://github.com/mozilla/hubs/discussions/2813

**Reply by DurianBoomb** [reply]
THANKS！！！！！！！

**Reply by DurianBoomb** [reply]
I read that discussion, and I have a client ran at my PC, but I still don't know where I should modified, can you give me some advice?

**Reply by johnshaughnessy** [reply]
@DurianBoomb We do not have guides/documentation for modifying the client, and it'll depend on what you're trying to do, but if you are just looking for somewhere to run some javascript every frame, one option is to put it in hubs-systems: https://github.com/mozilla/hubs/blob/master/src/systems/hubs-systems.js
(Or you could wrap your code in your own aframe components/systems)

**Reply by DurianBoomb** [reply]
> @DurianBoomb We do not have guides/documentation for modifying the client, and it'll depend on what you're trying to do, but if you are just looking for somewhere to run some javascript every frame, one option is to put it in hubs-systems: https://github.com/mozilla/hubs/blob/master/src/systems/hubs-systems.js
> (Or you could wrap your code in your own aframe components/systems)

I want to add my 3js into the scene, I'm new to front-end, can you give me further advice? Much appriciate!
